pkg/operator/ceph/object/notification/obc_label_controller_test.go

```diff
@@ -263,6 +292,146 @@ func TestCephBucketNotificationOBCLabelController(t *testing.T) {
 		res, err := r.Reconcile(ctx, req)
 		assert.NoError(t, err)
 		assert.False(t, res.Requeue)
-		assert.True(t, createWasInvoked)
+		assert.True(t, getWasInvoked)
+		assert.Equal(t, 1, createNotificationCount)
+		assert.Equal(t, 0, deleteNotificationCount)
+	})
+
+	t.Run("reconcile with already existing label for the obc", func(t *testing.T) {
+		resetValues()
+		noChangeOBC, noChangeOB := createOBResources(noChangeBucketName)
+		noChangeOBC.Labels = setNotificationLabels([]string{testNotificationName})
+		noChangeOBC.Spec.ObjectBucketName = noChangeBucketName
+		noChangeOBC.Status.Phase = bktv1alpha1.ObjectBucketClaimStatusPhaseBound
+		req.NamespacedName.Name = noChangeBucketName
+		objects := []runtime.Object{
+			cephCluster,
+			noChangeOBC,
+			noChangeOB,
+			bucketNotification,
+			bucketTopic,
+		}
+
+		cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(objects...).Build()
+
+		r := &ReconcileOBCLabels{client: cl, context: c, opManagerContext: ctx}
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.False(t, res.Requeue)
+		assert.True(t, getWasInvoked)
+		assert.Equal(t, 0, createNotificationCount)
```

Comment:
we should have one "create" here - we should reprovision the notification for the case that its configuration was changed